src/platform/packages/private/kbn-global-redirect-app-links/global_redirect_app_links.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { ReactNode, useEffect } from 'react';
+import { getClosestLink } from '@kbn/shared-ux-utility';
+
+export interface GlobalRedirectAppLinkProps {
+  children: ReactNode;
+  navigateToUrl: (url: string) => Promise<void> | void;
+}
+
+export const GlobalRedirectAppLink = ({ children, navigateToUrl }: GlobalRedirectAppLinkProps) => {
+  useEffect(() => {
+    function clickHandler(event: MouseEvent) {
+      const target = event.target as HTMLElement;
+
+      const link = getClosestLink(target);
```

Comment:
Heaven help us if it's more than 10 nodes away, right?

Reply:
right, that would trigger the full page reload (which is also not the end of the world). we could increase to 15 or 20 or remove :) 

In the current implementation we don't have any hard limit, which I thought is a bit annoying, because let's say you have `<RedirectAppLinks/>` on the application root which is very close to the <body/>. This means that every click in such app iterates to almost the very top. Almost doesn't matter if it is app root or <body/>. 

I thought that adding here some hard limit is just.

---

that being said, I just clicked around discover and dashboard and the worst element nesting (any element, not just linked) I could find is 32. So this might not even be a problem an I should remove the hard limit